blog/2024-07-07-openhab-4-2-release.md

```diff
@@ -0,0 +1,415 @@
+
+### `oh-context` Component
+
+_Justin Georgi ([@justingeorgi](https://github.com/JustinGeorgi)), openHAB Contributor_
+
+With openHAB 4.2, advanced custom widgets have gotten more powerful with the new `oh-context` component!
+
+This component allows widget authors to insert a code context block at any point in the widget tree and define three different types of objects available for use in expressions in all children of the `oh-context`:
+
+- Variables: These context variables differ slightly from the normal widget variables in that they can have initial default values and also have complete communication with sub-widgets.
+- Constants: Context constants make for much more concise and readable expressions but unlike variable are only evaluated once when the widget loads.
+- Functions: Named context functions allow widget authors to efficiently reuse complex functions in multiple expressions.
+
+Please have a look at the [documentation](/docs/ui/widget-expressions-variables.html) to learn about the new `oh-context` component.
+
+### Health UI
+
+Ever heard of orphaned links before? Orphaned links are channel links, where either the Item or the channel (or possibly even both) are missing.
+
+As you can imagine, orphaned links can and should be removed.
+The new health UI allows to easily check if there are orphaned links and take care of them:
+
+<p align="center"><img style="max-width: 70%;" src="/uploads/2024-07-07-openhab-4-2-release/health-menu.png"/></p>
+<p align="center"><img style="max-width: 70%;" src="/uploads/2024-07-07-openhab-4-2-release/health-orphan-links.png"/></p>
+
+The health UI is not only meant to display orphaned links, but also to provide a central place for health checks and maintenance tasks.
+So stay tuned what gets added in the future!
+
+Thanks to Arne Seime ([@seime](https://github.com/seime)) for this nice addition!
+
+### Blockly Enhancements
+
+_Stefan Höhn ([@stefan-hoehn](https://github.com/stefan-hoehn)), Blockly Maintainer_
+
+## Sitemap Enhancements
+
+_Laurent Garnier ([@lolodomo](https://github.com/lolodomo)), openHAB Maintainer_
+
+### Two possible behaviors for the Slider element
+
+A new parameter named `releaseOnly` was added to the `Slider` element.
+If `releaseOnly` parameter is specified, the new value is only sent on releasing the slider.
+Otherwise values are sent while sliding which is now the default behavior.
+
+Note: an implicit slider element (when `Default` element is used) will have the `releaseOnly` parameter set.
+Its behavior is therefore to send the new value only when the slider is released.
+
+### Press and release buttons
+
+For a `Switch` element with mappings, each button can now behave either as a click button or as a press and release button.
+The default behavior is a click button.
+To get a press and release behavior, provide two commands separated by a semicolon.
```

Comment:
@lolodomo This sentence says semicolon, but the example below has a colon. Which one is correct?

Reply:
Arghhh !
You're right, good catch , I made the confusion in the docs first, due to my limited English.
I will prepare a doc fix.

Reply:
https://github.com/openhab/openhab-docs/pull/2329